src/utils/httpClient.ts

```diff
@@ -0,0 +1,36 @@
+import axios from 'axios'
+
+export type AxiosDNSLookupFunction = typeof axios.defaults.lookup
+
+export const httpClient = axios.create({})
+
+export const sendAxiosRequestWithRequestInit = async (url: string | URL, requestInit: RequestInit) => {
+  const { method, headers, ...rest } = requestInit
+
+  if (url instanceof URL) {
```

Comment:
In all our usages, it seems we are providing the url as a `URL` type. Therefore, we could specify the type as `URL` instead of `string | URL`. I'm in favor of making this adjustment when `string` is going to be used, but I'll leave this decision up to you.

Reply:
I agree, let's remove the type `string` from `sendAxiosRequestWithRequestInit` method `url` argument, let it be `url: URL`